stages/01_find_data.py

```diff
@@ -0,0 +1,27 @@
+# Imports
+# -------
+import urllib.request
+from bs4 import BeautifulSoup
+
+dude_target_page = urllib.request.urlopen("https://dude.docking.org/targets")
+dude_target_page = dude_target_page.read().decode('utf-8')
+
+soup = BeautifulSoup(dude_target_page, 'lxml')
+target_table = soup.find('table')
+rows = target_table.findAll('tr')
+
+active_base_url = 'https://dude.docking.org//targets/PROTEIN/actives_nM_combined.ism\n'
+inactive_base_url = 'https://dude.docking.org//targets/PROTEIN/inactives_nM_combined.ism\n'
```

Comment:
Why this approach to get the links by outputting the `wget` lines?

You can do something like the following:

```shell
$ perl -Mojo -E 'say g(shift)->dom("tr td:nth-child(2) a")->map("attr", "href")->join("\n")' 'https://dude.docking.org/targets'
https://dude.docking.org/targets/aa2ar
https://dude.docking.org/targets/abl1
https://dude.docking.org/targets/ace
https://dude.docking.org/targets/aces
https://dude.docking.org/targets/ada
https://dude.docking.org/targets/ada17
# ...
```

or

```python
import pandas as pd

url = 'https://dude.docking.org/targets';
df = pd.read_html(url)[0]
df['Target Name'].to_list()

# ['AA2AR',
#  'ABL1',
#  'ACE',
#  'ACES',
#  'ADA',
#  'ADA17',
#  'ADRB1',
#  'ADRB2',
#  'AKT1',
#  'AKT2',
#  'ALDR',
#  'AMPC',
#  ...
# ]
```

and build the URLs from that.

Reply:
It would be fine for `download` and `brick` to contain multiple levels for directories. The files do not have to all be at a single level. In fact, when dealing with many, many files, using directories is faster. This can be done using only `wget`.

Reply:
Oh! I didn't realize that with pandas. I can do that approach. I have worked with perl once a decade ago so I am not familiar with the language.